x-pack/plugins/observability/public/utils/slo/convert_sli_apm_params_to_apm_app_deeplink_url.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+interface Props {
+  duration: string;
+  environment: string;
+  filter: string | undefined;
+  service: string;
+  transactionName: string;
+  transactionType: string;
+}
+
+export function convertSliApmParamsToApmAppDeeplinkUrl({
+  duration,
+  environment,
+  filter,
+  service,
+  transactionName,
+  transactionType,
+}: Props) {
+  if (!service) {
+    return '';
+  }
+
+  const qs = new URLSearchParams('comparisonEnabled=true');
```

Comment:
📣  used URLSearchParams to simplify the url build part

Reply:
Nice! Very clean.

Reply:
Cool service, I don't know If APM registered with it, I'll take a look, thanks!